logstash-core-event-java/src/main/java/org/logstash/Event.java

```diff
@@ -76,9 +76,19 @@ public Event(Map data)
         this.metadata_accessors = new Accessors(this.metadata);
 
         this.cancelled = false;
-        this.timestamp = initTimestamp(data.get(TIMESTAMP));
+
+        Object provided_timestamp = data.get(TIMESTAMP);
+        Timestamp parsed_timestamp = initTimestamp(provided_timestamp);
+        this.timestamp = (parsed_timestamp == null) ? Timestamp.now() : parsed_timestamp;
+
         this.data.put(TIMESTAMP, this.timestamp);
         this.accessors = new Accessors(this.data);
+
+        // the tag() method has to be called after the Accessors initialization
+        if (parsed_timestamp == null) {
```

Comment:
how is this ever `null` if line 82 checks for this and then sets it to a non-null value